src/mds/QuiesceDbManager.cc

```diff
@@ -52,11 +52,25 @@ static QuiesceTimeInterval time_distance(QuiesceTimePoint lhs, QuiesceTimePoint
   }
 }
 
-bool QuiesceDbManager::db_thread_has_work() const
+bool QuiesceDbManager::db_thread_has_work(bool leader_bootstrapping) const
 {
+  if (leader_bootstrapping) {
+    // if the leader is bootstrapping, `db_thread_should_exit` cannot be true
+    ceph_assert(!db_thread_should_exit);
+    if ((agent_callback.has_value() && agent_callback->if_newer < db.version())
+    || (cluster_membership.has_value() && cluster_membership->epoch != membership.epoch)) {
+      return true;
+    }
+    // if (pending_db_updates.size() > 0 || pending_acks.size() > 0
+    // || pending_requests.size() > 0)
+    // if the the requests/acks aren't empty then skipping a sleep will cause
+    // a busy-loop of the quiesce manager thread.
+    return false;
+  }
+
+  // if the leader isn't bootstrapping, pending acks/reqs should be empty
+  ceph_assert(pending_acks.empty() && pending_requests.empty());
   return db_thread_should_exit
-      || pending_acks.size() > 0
-      || pending_requests.size() > 0
       || pending_db_updates.size() > 0
```

Comment:
im still confused about `pending_db_updates`

Reply:
Thanks for the tracker @dparmar18 - I'll plant to go through the details in the tracker.